resources/qml/Avatar.qml

```diff
@@ -35,10 +36,29 @@ Rectangle {
         font.pixelSize: avatar.height / 2
         verticalAlignment: Text.AlignVCenter
         horizontalAlignment: Text.AlignHCenter
-        visible: img.status != Image.Ready
+        visible: img.status != Image.Ready && !Settings.useIdenticon
         color: Nheko.colors.text
     }
 
+    Image {
+        id: identicon
+        anchors.fill: parent
+        visible: img.status != Image.Ready && Settings.useIdenticon
+        source: Settings.useIdenticon ? ("image://jdenticon/" + (userid !== "" ? userid : roomid) + "?radius=" + (Settings.avatarCircles ? 100 : 25) + ((avatar.crop) ? "" : "&scale")) : ""
```

Comment:
You can probably drop the scaling, jdenticons are always square, are they not? In any case I would always scale them and never crop.